File: Typecast/TCGlyphListViewController.m
//
//  TCGlyphListViewController.m
//  Type Designer
//
//  Created by David Schweinsberg on 16/12/12.
//  Copyright (c) 2012 David Schweinsberg. All rights reserved.
//

#import "TCGlyphListViewController.h"
#import "TCDocument.h"
#import "TCGlyphWindowController.h"

@interface TCGlyphListViewController ()

@end

@implementation TCGlyphListViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Initialization code here.
    }
    
    return self;
}

- (void)showGlyphs:(NSArray *)glyphs
{
    TCGlyphWindowController *windowController = [[TCGlyphWindowController alloc] initWithWindowNibName:@"GlyphWindow"];
    [[self document] addWindowController:windowController];
    [windowController setGlyphDescription:[glyphs lastObject]];
    [windowController showWindow:self];
}

@end

File: Typecast/TCGlyphListViewController.swift
//
//  TCGlyphListViewController.swift
//  Type Designer
//
//  Created by David Schweinsberg on 8/7/17.
//  Copyright © 2017 David Schweinsberg. All rights reserved.
//

import Cocoa

class TCGlyphListViewController: NSViewController, NSTableViewDelegate,
    NSTableViewDataSource {
  @IBOutlet weak var tableView: NSTableView?

  override func viewDidLoad() {
    super.viewDidLoad()
      // Do view setup here.
  }

  func numberOfRows(in tableView: NSTableView) -> Int {
    if let (_, table) = representedObject as? (FontDocument, TCGlyfTable) {
      return table.descript.count
    } else {
      return 0
    }
  }

  func tableView(_ tableView: NSTableView, viewFor tableColumn: NSTableColumn?, row: Int) -> NSView? {
    if let (_, table) = representedObject as? (FontDocument, TCGlyfTable),
      let tableColumn = tableColumn {
      if tableColumn.identifier.rawValue == "GlyphIndex",
        let view = tableView.makeView(withIdentifier: NSUserInterfaceItemIdentifier(rawValue: "GlyphIndex"), owner: self) as? NSTableCellView {
        let descript = table.description(at: row)
        view.textField?.stringValue = String(descript.glyphIndex)
        return view
      } else if tableColumn.identifier.rawValue == "GlyphName",
        let view = tableView.makeView(withIdentifier: NSUserInterfaceItemIdentifier(rawValue: "GlyphName"), owner: self) as? NSTableCellView {
        let descript = table.description(at: row)
//        view.textField?.stringValue = descript.name
        return view
      }
    }
    return nil
  }

  @IBAction func showGlyphWindow(_ sender: Any?) {
    if let (document, _) = representedObject as? (FontDocument, TCGlyfTable),
      let tableView = tableView,
      tableView.selectedRow >= 0 {
    }
  }
}
